pkg/pool-stable/contracts/ComposableStablePool.sol

```diff
@@ -305,6 +305,13 @@ contract ComposableStablePool is
         uint256 registeredIndexOut,
         uint256[] memory scalingFactors
     ) private returns (uint256) {
+        (
+            uint256 preJoinExitSupply,
+            uint256[] memory balances,
+            uint256 currentAmp,
+            uint256 preJoinExitInvariant
+        ) = _beforeJoinExit(registeredBalances);
+
         bool isGivenIn = swapRequest.kind == IVault.SwapKind.GIVEN_IN;
 
         _upscaleArray(registeredBalances, scalingFactors);
```

Comment:
This is wrong :( We need to upscale the `registeredBalances` _before_ calling `_beforeJoinExit`.

Reply:
This doesn't happen on the join/exit functions since we have a layer there that deals with scaling, but we're manually doing all of that here.